aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -0,0 +1,180 @@
+                                 List<AttributeDefinition> definitions) {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(tableName)
+          .build());
+    } catch (ResourceNotFoundException e) {
+      LOG.info("Glue lock DynamoDB table <{}> not found, try to create", tableName);
+      dynamo.createTable(CreateTableRequest.builder()
+          .tableName(tableName)
+          .keySchema(schema)
+          .attributeDefinitions(definitions)
+          .billingMode(BillingMode.PAY_PER_REQUEST)
+          .build());
+
+      boolean isTableActive = false;
+      while (!isTableActive) {
+        LOG.info("waiting for DynamoDB table <{}> to be active", tableName);
+        try {
+          Thread.sleep(5000);
+        } catch (InterruptedException ie) {
+          LOG.warn("Glue lock DynamoDB table creation sleep interrupted", e);
+        }
+        DescribeTableResponse describeTableResponse = dynamo.describeTable(DescribeTableRequest.builder()
+            .tableName(tableName)
+            .build());
+        isTableActive = describeTableResponse.table().tableStatus().equals(TableStatus.ACTIVE);
+      }
+    }
+  }
+
+
+  @Override
+  public boolean tryLock(String database, String table, long expireMillis) {
+    String tableId = tableId(database, table);
+
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put(LOCK_TABLE_COL_TABLE_ID, AttributeValue.builder().s(tableId).build());
+    key.put(LOCK_TABLE_COL_EXPIRE_TS_MILLIS, AttributeValue.builder().n(
+        Long.toString(System.currentTimeMillis() + expireMillis)
+    ).build());
+
+    Map<String, AttributeValue> expressionValues = Maps.newHashMap();
+    expressionValues.put(":tid", AttributeValue.builder().s(tableId).build());
+    expressionValues.put(":ts", AttributeValue.builder().n(Long.toString(System.currentTimeMillis())).build());
+
+    try {
+
+      dynamo.putItem(PutItemRequest.builder()
+          .tableName(awsProperties.glueCatalogLockTable())
+          .item(key)
+          // succeed only if there is no lock, or the lock is there but already expired
+          .conditionExpression("attribute_not_exists(" +
+              LOCK_TABLE_COL_TABLE_ID + ") OR (" +
+              LOCK_TABLE_COL_TABLE_ID + " = :tid AND " +
+              LOCK_TABLE_COL_EXPIRE_TS_MILLIS + " < :ts)")
+          .expressionAttributeValues(expressionValues)
+          .build());
+      return true;
+    } catch (Exception e) {
+      // most likely it's ConditionalCheckFailedException, but we will catch any exception
+      LOG.debug("Acquiring lock {}.{} failed", database, table, e);
+      return false;
+    }
+  }
+
+  @Override
+  public void unlock(String database, String table) {
+    String tableId = tableId(database, table);
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put(LOCK_TABLE_COL_TABLE_ID, AttributeValue.builder().s(tableId).build());
+
+    dynamo.deleteItem(DeleteItemRequest.builder()
```

Comment:
add log line.

We should inform if the item does not exist in the table.

Reply:
what log line do you mean? It does not throw exception or have any indication if it is not in the table.

Reply:
I think we need to do a conditional delete to ensure that this does not release the lock held by another process. As long as that requires a conditional delete, it would make sense to add `attribute_exists` to ensure that this only succeeds if the lock exists and was still held by this process. Then a failure of either assumption will result in `ConditionalCheckFailedException` and this can log that the lock was no longer held.